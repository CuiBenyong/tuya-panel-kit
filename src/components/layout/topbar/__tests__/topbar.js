import 'react-native';
import _ from 'lodash';
import React from 'react';
import { Platform } from 'react-native';
import renderer from 'react-test-renderer';
import TopBar from '../index';

const backIcon = Platform.OS === 'ios' ? 'backIos' : 'backAndroid';

describe('TopBar Component', () => {
  it('basic render', () => {
    const component = renderer
      .create(
        <TopBar.Container background="#000">
          <TopBar.Action name={backIcon} color="red" onPress={() => {}} />
          <TopBar.Content title="Title" />
          <TopBar.Action name="pen" color="yellow" onPress={() => {}} />
        </TopBar.Container>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('render gradient', () => {
    const linearBackground = {
      stops: {
        '0%': 'red',
        '100%': 'yellow',
      },
    };
    const radialBackground = {
      stops: [
        {
          offset: '0%',
          stopColor: '#ff0',
          stopOpacity: '1',
        },
        {
          offset: '100%',
          stopColor: '#00f',
          stopOpacity: '1',
        },
      ],
    };
    const Linear = renderer
      .create(
        <TopBar.Container background={linearBackground}>
          <TopBar.Action name={backIcon} onPress={() => {}} />
          <TopBar.Content title="Title" />
        </TopBar.Container>
      )
      .toJSON();
    const Radial = renderer.create(
      <TopBar
        style={{ marginTop: 24 }}
        background={radialBackground}
        title="Title"
        onBack={() => {}}
      />
    );
    expect(Linear).toMatchSnapshot();
    expect(Radial).toMatchSnapshot();
  });

  it('render multiple actions', () => {
    const linearBackground = {
      stops: {
        '0%': 'red',
        '100%': 'yellow',
      },
    };
    const radialBackground = {
      stops: [
        {
          offset: '0%',
          stopColor: '#ff0',
          stopOpacity: '1',
        },
        {
          offset: '100%',
          stopColor: '#00f',
          stopOpacity: '1',
        },
      ],
    };
    const C1 = renderer
      .create(
        <TopBar.Container background="blue">
          <TopBar.Action name={backIcon} onPress={() => {}} />
          <TopBar.Action source="定时" color="red" onPress={() => {}} />
          <TopBar.Content
            title="Very Very Very Very Very Long Title"
            subTitle="SubTitle"
            onPress={() => {}}
          />
          {['plus', 'warning', 'edit'].map(v => (
            <TopBar.Action key={v} name={v} onPress={() => {}} />
          ))}
        </TopBar.Container>
      )
      .toJSON();
    const C2 = renderer.create(
      <TopBar
        style={{ marginTop: 24 }}
        background="blue"
        title="Very Very Very Very Very Long Title"
        subTitle="SubTitle"
        onPress={() => {}}
        leftActions={[
          {
            name: backIcon,
            onPress: () => {},
          },
          {
            source: '定时',
            color: 'red',
            onPress: () => {},
          },
        ]}
        actions={['plus', 'warning', 'edit'].map(v => ({
          name: v,
          onPress: () => {},
        }))}
      />
    );
    expect(C1).toMatchSnapshot();
    expect(C2).toMatchSnapshot();
  });
});
