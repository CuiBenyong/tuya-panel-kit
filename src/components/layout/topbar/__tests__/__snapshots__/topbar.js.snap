// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar Component basic render 1`] = `
<View
  style={
    Array [
      null,
      Object {
        "height": 88,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 44,
        "justifyContent": "space-between",
        "marginTop": 44,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 44,
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingLeft": 12,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "marginRight": 4,
              "tintColor": "#fff",
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 16,
                },
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#fff",
                    "fontSize": 16,
                  },
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#fff",
                    "fontSize": 16,
                  },
                ],
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 44,
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingRight": 12,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Object {
              "marginLeft": 4,
              "tintColor": "#fff",
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`TopBar Component basic render 2`] = `
<View
  style={
    Array [
      null,
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 56,
        "justifyContent": "space-between",
      },
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 56,
          "overflow": "hidden",
          "paddingLeft": 20,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "marginRight": 10,
            "tintColor": "#fff",
            "width": 16,
          }
        }
      />
      <View
        style={
          Array [
            Object {},
            Object {
              "marginRight": 20,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 16,
              },
              Object {
                "backgroundColor": "transparent",
                "color": "#fff",
                "fontSize": 18,
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "height": 56,
          "justifyContent": "center",
          "width": 610,
        },
        Object {},
        false,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 16,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#fff",
              "fontSize": 18,
            },
            null,
          ],
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 56,
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingRight": 20,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "marginLeft": 10,
            "tintColor": "#fff",
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`TopBar Component should render only with centerText 1`] = `
<View
  style={
    Array [
      null,
      Object {
        "height": 88,
      },
    ]
  }
>
  <View
    pointerEvents="none"
    style={
      Object {
        "alignItems": "center",
        "height": 44,
        "justifyContent": "center",
        "left": 70,
        "position": "absolute",
        "right": 70,
        "top": 44,
        "width": 610,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 16,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#fff",
              "fontSize": 16,
            },
            null,
          ],
        ]
      }
    >
      Title
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 44,
        "justifyContent": "space-between",
        "marginTop": 44,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 44,
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingLeft": 12,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "marginRight": 4,
              "tintColor": "#fff",
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 16,
                },
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#fff",
                    "fontSize": 16,
                  },
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#fff",
                    "fontSize": 16,
                  },
                ],
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 44,
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingRight": 12,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Object {
              "marginLeft": 4,
              "tintColor": "#fff",
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`TopBar Component should render only with centerText 2`] = `
<View
  style={
    Array [
      null,
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 56,
        "justifyContent": "space-between",
      },
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 56,
          "overflow": "hidden",
          "paddingLeft": 20,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "marginRight": 10,
            "tintColor": "#fff",
            "width": 16,
          }
        }
      />
      <View
        style={
          Array [
            Object {},
            Object {
              "marginRight": 20,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 16,
              },
              Object {
                "backgroundColor": "transparent",
                "color": "#fff",
                "fontSize": 18,
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "height": 56,
          "justifyContent": "center",
          "width": 610,
        },
        Object {},
        false,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 16,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#fff",
              "fontSize": 18,
            },
            null,
          ],
        ]
      }
    >
      Title
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 56,
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingRight": 20,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "marginLeft": 10,
            "tintColor": "#fff",
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`TopBar Component should render only with leftImage 1`] = `
<View
  style={
    Array [
      null,
      Object {
        "height": 88,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 44,
        "justifyContent": "space-between",
        "marginTop": 44,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 44,
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingLeft": 12,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "marginRight": 4,
              "tintColor": "#fff",
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 16,
                },
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#fff",
                    "fontSize": 16,
                  },
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#fff",
                    "fontSize": 16,
                  },
                ],
              ]
            }
          />
        </View>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 44,
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingRight": 12,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Object {
              "marginLeft": 4,
              "tintColor": "#fff",
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`TopBar Component should render only with leftImage 2`] = `
<View
  style={
    Array [
      null,
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 56,
        "justifyContent": "space-between",
      },
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 56,
          "overflow": "hidden",
          "paddingLeft": 20,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "marginRight": 10,
            "tintColor": "#fff",
            "width": 16,
          }
        }
      />
    </View>
  </View>
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "height": 56,
          "justifyContent": "center",
          "width": 610,
        },
        Object {},
        false,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 16,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#fff",
              "fontSize": 18,
            },
            null,
          ],
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 56,
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingRight": 20,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "marginLeft": 10,
            "tintColor": "#fff",
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`TopBar Component should render with rightImage and rightText 1`] = `
<View
  style={
    Array [
      null,
      Object {
        "height": 88,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 44,
        "justifyContent": "space-between",
        "marginTop": 44,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 44,
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingLeft": 12,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "marginRight": 4,
              "tintColor": "#fff",
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 16,
                },
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#fff",
                    "fontSize": 16,
                  },
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#fff",
                    "fontSize": 16,
                  },
                ],
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 44,
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingRight": 12,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 16,
                },
                Object {
                  "backgroundColor": "transparent",
                  "color": "#fff",
                  "fontSize": 16,
                },
              ]
            }
          >
            Back
          </Text>
        </View>
        <Image
          resizeMode="contain"
          source={1}
          style={
            Object {
              "marginLeft": 4,
              "tintColor": "#fff",
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`TopBar Component should render with rightImage and rightText 2`] = `
<View
  style={
    Array [
      null,
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 56,
        "justifyContent": "space-between",
      },
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 56,
          "overflow": "hidden",
          "paddingLeft": 20,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "marginRight": 10,
            "tintColor": "#fff",
            "width": 16,
          }
        }
      />
      <View
        style={
          Array [
            Object {},
            Object {
              "marginRight": 20,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 16,
              },
              Object {
                "backgroundColor": "transparent",
                "color": "#fff",
                "fontSize": 18,
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "height": 56,
          "justifyContent": "center",
          "width": 610,
        },
        Object {},
        false,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 16,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#fff",
              "fontSize": 18,
            },
            null,
          ],
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 56,
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingRight": 20,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
            Object {
              "marginLeft": 20,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 16,
              },
              Object {
                "backgroundColor": "transparent",
                "color": "#fff",
                "fontSize": 18,
              },
            ]
          }
        >
          Back
        </Text>
      </View>
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "marginLeft": 10,
            "tintColor": "#fff",
          }
        }
      />
    </View>
  </View>
</View>
`;
